File: app/src/main/java/com/futnote/mobile/viewmodels/NoteViewModel.java
package com.futnote.mobile.viewmodels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.futnote.mobile.models.Note;
import com.futnote.mobile.models.NoteRepository;

import java.util.List;

public class NoteViewModel extends AndroidViewModel {

    private NoteRepository noteRepository;
    private LiveData<List<Note>> records;

    public NoteViewModel(@NonNull Application application) {
        super(application);
        noteRepository = new NoteRepository(application);
        records = noteRepository.select();
    }

    public void insert(Note note) {
        noteRepository.insert(note);
    }

    public void update(Note note) {
        noteRepository.update(note);
    }

    public void delete(Note note) {
        noteRepository.delete(note);
    }

    public void deleteAll() {
        noteRepository.deleteAll();
    }

    public LiveData<List<Note>> select() {
        return records;
    }
}

File: app/src/main/java/com/futnote/mobile/viewmodel/NoteViewModel.kt
package com.futnote.mobile.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.futnote.mobile.model.Note
import com.futnote.mobile.model.NoteRepository

class NoteViewModel(application: Application) : AndroidViewModel(application) {
    private var noteRepository: NoteRepository
    private var records: LiveData<List<Note>>

    fun insert(note: Note?) {
        noteRepository.insert(note)
    }

    fun update(note: Note?) {
        noteRepository.update(note)
    }

    fun delete(note: Note?) {
        noteRepository.delete(note)
    }

    fun select(): LiveData<List<Note>> {
        return records
    }

    init {
        noteRepository = NoteRepository(application)
        records = noteRepository.select()
    }
}